utils/check_if_new_model_added.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2024 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This script is used to get the directory of the modeling file that is added in a pull request (i.e. a new model PR).
+
+Usage:
+
+```bash
+python utils/check_if_new_model_added.py
+```
+"""
+
+import re
+from pathlib import Path
+from typing import List
+
+from git import Repo
+
+
+PATH_TO_REPO = Path(__file__).parent.parent.resolve()
+
+
+def get_new_files_between_commits(base_commit: str, commits: List[str]) -> List[str]:
+    """
+    Get the list of added python files between a base commit and one or several commits.
+
+    Args:
+        repo (`git.Repo`):
+            A git repository (for instance the Transformers repo).
+        base_commit (`str`):
+            The commit reference of where to compare for the diff. This is the current commit, not the branching point!
+        commits (`List[str]`):
+            The list of commits with which to compare the repo at `base_commit` (so the branching point).
+
+    Returns:
+        `List[str]`: The list of python files added between a base commit and one or several commits.
+    """
+    code_diff = []
+    for commit in commits:
+        for diff_obj in commit.diff(base_commit):
+            # We always add new python files
+            if diff_obj.change_type == "A" and diff_obj.b_path.endswith(".py"):
```

Comment:
Do we want to just constrain to adding new models? If someone is touching modeling code for e.g. llama, it's important those slow tests are run too. 

Maybe just asking people to run with `[run-slow]` in the commit message it enough for these cases, but it would for them to be picked up

Reply:
The problem of other cases is: they might contain multiple files touched and take too long to run. For example, a pull could actually trigger the whole CI. In fact, a new model PR would trigger the whole CI too  as it touch `transformers.__init__`.

The new workflow file added in this PR has a modification:  it runs a single model (as the new model PR has a well-defined target model to run: it's the new model to be test against).

In general cases, if we trigger all modified modeling files - it might take too long (and possibly from multiple PRs)

Reply:
OK - can we still enable running with a `run_slow` commit flag in a follow-up, such that we can request if and when it's needed

Reply:
> OK - can we still enable running with a run_slow commit flag in a follow-up, such that we can request if and when it's needed

For this, do  you mean run all tests (even slow ones) that are detected by the test fetcher?